File: src/main/java/com/mbayou/kick4k/authorization/FileRefreshTokenStore.java
package com.mbayou.kick4k.authorization;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class FileRefreshTokenStore implements RefreshTokenStore {
    private final Path path;

    public FileRefreshTokenStore(Path path) {
        this.path = path;
    }

    @Override
    public String getRefreshToken() {
        try {
            return Files.readString(this.path);
        } catch (IOException exception) {
            throw new RuntimeException("Couldn't read refresh token", exception);
        }
    }

    @Override
    public void notifyRefreshTokenRoll(String newRefreshToken) {
        try {
            Files.writeString(this.path, newRefreshToken);
        } catch (IOException exception) {
            throw new RuntimeException("Couldn't write refresh token", exception);
        }
    }
}

File: src/main/kotlin/com/mbayou/kick4k/authorization/FileRefreshTokenStore.kt
package com.mbayou.kick4k.authorization

import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

class FileRefreshTokenStore(private val path: Path) : RefreshTokenStore {
    override fun getRefreshToken(): String {
        return try {
            Files.readString(path)
        } catch (exception: IOException) {
            throw RuntimeException("Couldn't read refresh token", exception)
        }
    }

    override fun notifyRefreshTokenRoll(newRefreshToken: String) {
        try {
            Files.writeString(path, newRefreshToken)
        } catch (exception: IOException) {
            throw RuntimeException("Couldn't write refresh token", exception)
        }
    }
}
